usaspending_api/etl/transaction_loaders/fpds_loader.py

```diff
@@ -0,0 +1,289 @@
+
+
+def delete_stale_fpds(date):
+    """
+    Removed transaction_fpds and transaction_normalized records matching any of the
+    provided detached_award_procurement_id list
+    Returns list of awards touched
+    """
+    if not date:
+        return []
+
+    detached_award_procurement_ids = get_deleted_fpds_data_from_s3(date)
+
+    if detached_award_procurement_ids:
+        with psycopg2.connect(dsn=USASPENDING_CONNECTION_STRING) as connection:
+            with connection.cursor() as cursor:
+                cursor.execute(
+                    "select transaction_id from transaction_fpds where detached_award_procurement_id in ({})".format(
+                        ",".join([str(id) for id in detached_award_procurement_ids])
+                    )
+                )
+                # assumes, possibly dangerously, that this won't be too many for the job to handle
+                transaction_normalized_ids = cursor.fetchall()
+
+                # Set backreferences from Awards to Transaction Normalized to null. These pointers will be correctly updated
+                # in the update awards stage later on
+                cursor.execute(
+                    "update awards set latest_transaction_id = null, earliest_transaction_id = null "
+                    "where latest_transaction_id in ({}) returning id".format(
+                        ",".join([str(row[0]) for row in transaction_normalized_ids])
+                    )
+                )
+                awards_touched = cursor.fetchall()
+
+                # Remove Trasaction FPDS rows
+                cursor.execute(
+                    "delete from transaction_fpds where detached_award_procurement_id in ({})".format(
+                        ",".join([str(id) for id in detached_award_procurement_ids])
+                    )
+                )
+
+                # Remove Transaction Normalized rows
+                cursor.execute(
+                    "delete from transaction_normalized where id in ({})".format(
+                        ",".join([str(row[0]) for row in transaction_normalized_ids])
+                    )
+                )
+
+                return awards_touched
+    else:
+        return []
+
+
+def load_ids(chunk):
+    """
+    Run transaction load for the provided ids. This will create any new rows in other tables to support the transaction
+    data, but does NOT update "secondary" award values like total obligations or C -> D linkages. If transactions are
+    being reloaded, this will also leave behind rows in supporting tables that won't be removed unless destroy_orphans
+    is called.
+    returns ids for each award touched
+    """
+    with Timer() as timer:
+        retval = []
+        if chunk:
+            broker_transactions = _extract_broker_objects(chunk)
+            if broker_transactions:
+                load_objects = _transform_objects(broker_transactions)
+
+                retval = _load_transactions(load_objects)
```

Comment:
If I am reading this correct, this function is the "kernel" of the loader (Provide IDs, it will fetch from broker and store in USAspending)

Simply Simple!

Reply:
I'm not really sure what a kernel is outside the context of operating systems, but if you like it, sure! It's a kernel!